### PR TITLE
Add new testnet command to auto set up your node

### PR DESCRIPTION
**Pull request by NullSoldier**
## Summary
this new command will fetch a users test net user and auto configure
their node with their testnet user with their graffiti and node name.

## Testing Plan
Manually tested

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ironfish-cli/src/commands/testnet.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      userId = Number(userArg)
+    }
+
+    if (userId === null) {
+      this.log(`Could not figure out testnet user id from ${userArg}`)
+      this.exit(1)
+    }
+
+    // request user from API
+    this.log(`Asking Ironfish who user ${userId} is...`)
```
Comment (deekerno) [inline ironfish-cli/src/commands/testnet.ts:73]: Minor branding nit: Should be "Iron Fish".

**Review by deekerno — APPROVED** [review]
Manually tested, works on my end. Left one minor nit.

File: ironfish/src/webApi.ts [hunk 2]
```diff
@@ -100,6 +108,13 @@ export class WebApi {
     return response.data.data
   }
 
+  async getUser(id: number): Promise<ApiUser | null> {
+    return await axios
+      .get<ApiUser>(`${this.host}/users/${id}`, this.options())
+      .then((r) => r.data)
+      .catch(() => null)
```
Comment (brekk) [inline ironfish/src/webApi.ts:115]: Are there error cases which necessitate this line?

File: ironfish-cli/src/commands/testnet.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+export default class Testnet extends IronfishCommand {
+  static hidden = false
+  static description = 'Set up your node to mine for the testnet'
+
+  static flags = {
+    [VerboseFlagKey]: VerboseFlag,
+    [DataDirFlagKey]: DataDirFlag,
+    confirm: flags.boolean({
+      default: false,
+      description: 'confirm without asking',
```
Comment (brekk) [inline ironfish-cli/src/commands/testnet.ts:19]: Capitalization required?

File: ironfish-cli/src/commands/testnet.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+      description: "Don't update your graffiti",
+    }),
+  }
+
+  static args = [
+    {
+      name: 'user',
+      required: false,
+      description:
+        'the user id or url to a testnet user like https://testnet.ironfish.network/users/1080',
```
Comment (brekk) [inline ironfish-cli/src/commands/testnet.ts:36]: Same

File: ironfish-cli/src/commands/testnet.ts [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    let userArg = ((args.user as string | undefined) || '').trim()
+
+    if (!userArg) {
+      userArg = (await cli.prompt(
+        'Enter the user id or url to a testnet user like https://testnet.ironfish.network/users/1080\nUser ID or URL',
+        {
+          required: true,
+        },
+      )) as string
+      this.log('')
```
Comment (brekk) [inline ironfish-cli/src/commands/testnet.ts:51]: This is to print a newline?

File: ironfish-cli/src/commands/testnet.ts [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    let userId: number | null = null
```
Comment (brekk) [inline ironfish-cli/src/commands/testnet.ts:54]: Does TS require this construct?

File: ironfish-cli/src/commands/testnet.ts [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+    if (userArg.indexOf('/users/') !== -1) {
+      const index = userArg.indexOf('/users/')
```
Comment (brekk) [inline ironfish-cli/src/commands/testnet.ts:57]:
```suggestion
    const index = userArg.indexOf('/users/')
    if (index !== -1) {
```

File: ironfish/src/webApi.ts (hunk 2 above)
Comment (NullSoldier) [reply ironfish/src/webApi.ts:115]: If the user is not found you get a 404 which produces an exception

File: ironfish-cli/src/commands/testnet.ts (hunk 6 above)
Comment (NullSoldier) [reply ironfish-cli/src/commands/testnet.ts:54]: Some times, but not in this case. If ommitted the type is "materialized" at the first line that requires it be a certain type.

File: ironfish-cli/src/commands/testnet.ts (hunk 5 above)
Comment (NullSoldier) [reply ironfish-cli/src/commands/testnet.ts:51]: yes

File: ironfish-cli/src/commands/testnet.ts (hunk 3 above)
Comment (NullSoldier) [reply ironfish-cli/src/commands/testnet.ts:19]: which part? I don't think so.

File: ironfish-cli/src/commands/testnet.ts (hunk 1 above)
Comment (NullSoldier) [reply ironfish-cli/src/commands/testnet.ts]: fixed

**Pull request merged**